actions/kubernetes_helpers.go

```diff
@@ -65,27 +72,29 @@ func getNodeForPatching(ctx context.Context, log logrus.FieldLogger, clientset k
 	// on GKE we noticed that sometimes the node is not found, even though it is in the cluster
 	// as a result was returned from watch. But subsequent get request returns not found.
 	// This is likely due to clientset's caching that's meant to alleviate API's load.
-	// So we give enough time for cache to sync.
-	logRetry := func(err error, _ time.Duration) {
-		log.Warnf("getting node, will retry: %v", err)
-	}
+	// So we give enough time for cache to sync - ~10s max.
+
 	var node *v1.Node
-	b := backoff.NewExponentialBackOff()
-	b.MaxElapsedTime = 10 * time.Second
-	err := backoff.RetryNotify(func() error {
+
+	boff := waitext.WithMaxRetries(waitext.DefaultExponentialBackoff(), 5)
```

Comment:
I changed this because `wait` does not have a way to say "Stop after X duration" directly. It uses contexts for that but then here we would surface the context.Cancelled error and not the `Nodes().Get()` error, which breaks calling code.

I think this is an OK compromise, it still waits for ~10s tops and surfaces the inner error (unless the overall context is cancelled of course).

Reply:
> does not have a way to say "Stop after X duration" directly.

I think `Backoff.Cap` is used for this.

Reply:
See comment in this thread https://github.com/castai/cluster-controller/pull/113#discussion_r1577728079 

TL;DR Backoff.Cap works on a single wait interval, not the overall operation - context.Timeout should be used for that.